plugins/discourse-lazy-videos/lib/lazy-videos/lazy_vimeo.rb

```diff
@@ -8,16 +8,24 @@ class Onebox::Engine::VimeoOnebox
 
   def to_html
     if SiteSetting.lazy_videos_enabled && SiteSetting.lazy_vimeo_enabled
-      video_id = oembed_data[:video_id]
+      full_video_id = oembed_data[:uri].sub("/videos/", "").sub(":", "/")
+
+      if !oembed_data[:uri].match?(%r{videos/\d+:.+})
```

Comment:
Let's add a note here explaining that this check is for unlisted videos as context for the next developer that looks into this bit of code.